internal/git/gitsync_processor.go

```diff
@@ -1,19 +1,76 @@
 package git
 
 import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/storage/memory"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
-	numaplanenumaprojiov1 "github.com/numaproj-labs/numaplane/api/v1"
+	v1 "github.com/numaproj-labs/numaplane/api/v1"
 )
 
+const messageChanLength = 5
+
+type Message struct {
+	Updates bool
+	Err     error
+}
+
 type GitSyncProcessor struct {
+	gitSync     v1.GitSync
+	channels    map[string]chan Message
+	k8Client    client.Client
+	clusterName string
+}
+
+func isValidTag(tag string) bool {
+	match, _ := regexp.MatchString(`^[vV]?\d+\.\d+\.\d+$`, tag)
+	return match
```

Comment:
Actually, do we need to impose these constraints on tags? Do we not allow a Tag which doesn't match this convention? (if we decide not to, perhaps we should add another field to the RepositoryPath to describe which one it is)

Reply:
the problem would be the go git package ,it takes the reference as input and that can be the branch ,tag or commit hash so we need some kind of criteria to distinguish whether its a branch ,commit hash or tag @juliev0

Reply:
replied in [slack](https://intuit-teams.slack.com/archives/C05K82P3N9J/p1705422186374989)